plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -0,0 +1,232 @@
+
+            if add_member:
+                return add_member.username
+
+        except (gitlab.exceptions.GitlabCreateError) as e:
+            self._module.fail_json(
+                msg="Failed to add member to the Group, Group ID %s: %s" % (gitlab_group_id, e))
+
+    # remove user from a group
+    def remove_user_from_group(self, gitlab_user_id, gitlab_group_id):
+        try:
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+
+            group = self._gitlab.groups.get(gitlab_group_id)
+            group.members.delete(gitlab_user_id)
+
+        except (gitlab.exceptions.GitlabDeleteError) as e:
+            self._module.fail_json(
+                msg="Failed to remove member from GitLab group, ID %s: %s" % (gitlab_group_id, e))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            server_url=dict(type='str', required=True),
+            access_token=dict(type='str', required=True, no_log=True),
+            gitlab_group=dict(type='str', required=True),
+            gitlab_user=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            access_level=dict(type='str', required=False, choices=['guest', 'reporter', 'developer', 'maintainer', 'owner'])
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_PY_GITLAB:
+        module.fail_json(msg=missing_required_lib('python-gitlab', url='https://python-gitlab.readthedocs.io/en/stable/'), exception=GITLAB_IMP_ERR)
+
+    server_url = module.params['server_url']
+    access_token = module.params['access_token']
+    gitlab_group = module.params['gitlab_group']
+    gitlab_user = module.params['gitlab_user']
+    state = module.params['state']
+    access_level = module.params['access_level']
+
+    # convert access level string input to int
+    if access_level:
+        access_level_int = {
+            'guest': gitlab.GUEST_ACCESS,
+            'reporter': gitlab.REPORTER_ACCESS,
+            'developer': gitlab.DEVELOPER_ACCESS,
+            'maintainer': gitlab.MAINTAINER_ACCESS,
+            'owner': gitlab.OWNER_ACCESS
+        }
+
+        access_level = access_level_int[access_level]
+
+    # connect to gitlab server
+    try:
+        gl = gitlab.Gitlab(server_url, private_token=access_token)
+    except (gitlab.exceptions.GitlabAuthenticationError, gitlab.exceptions.GitlabGetError) as e:
+        module.fail_json(msg="Failed to authenticate to GitLab Server: %s: %s" % (server_url, e))
+
+    group = GitLabGroup(module, gl)
+
+    user_exists = group.does_user_exist(gitlab_user)
+    group_exists = group.does_group_exist(gitlab_group)
+
+    # group doesn't exist
+    if not group_exists:
+        module.fail_json(msg="group '%s' not found." % gitlab_group)
+
+    # user doesn't exist
+    if not user_exists:
+        module.fail_json(msg="user '%s' not found." % gitlab_user)
```

Comment:
It could make sense to ignore non-existing users if `state=absent`.
```suggestion
        if state == 'absent':
             module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
        else:
            module.fail_json(msg="user '%s' not found." % gitlab_user)
```

Reply:
@felixfontein, hmm, How about this? 

I have nested if to validate a couple of conditions:

1 - first it will test if the group exists. if it doesn't exist, playbook will fail and print a msg to the user. if group exists, will go to step 2.

2 - get group id and check if user exists. if user doesn't exist, the playbook will fail and print a msg to the user. if username exists, go step 3

3 - get user id and check if it is a member in the specified group.

4 - if user is not a member in the group and state is specified as present, it will validate if user specifies the access level and add him/her to the group. if access level is not specified, it will fail. 

5 - if user is not a member in the group and state is specified as absent, it will print a msg declaring no change to make on gitlab server.

6 - if user is a member in the group and state is specified as present, it will print a msg declaring no change to make on gitlab server.

7 - if user is a member in the group and state is specified as absent, it will remove the username from the group.



` 
if group_exists:
        # get group id
        gitlab_group_id = group_exists.id
        
        if user_exists:
            #get user id
            gitlab_user_id = user_exists.id
            is_user_a_member = group.is_user_a_member(gitlab_group_id, gitlab_user_id)
            
            # check if the user is a member in the group
            if not is_user_a_member:
                if state == 'present':
                    if not access_level:
                        module.fail_json(msg="An access level must be entered for adding member to GitLab group '%s'." % gitlab_user)
                    
                    # add user to the group
                    group.add_member_to_group(gitlab_user_id,gitlab_group_id,access_level)
                    module.exit_json(changed=True, result="Successfully added user to the group '%s'." % gitlab_user)

                else:
                    module.exit_json(changed=False, result="User is not a member in the group. No change to report '%s'." % gitlab_user)
            # in case that a user is a member
            else:
                if state == 'present':
                    module.exit_json(changed=False, result="User is already a member in the group. No change to report '%s'." % gitlab_user)
                else:
                    # remove the user from the group
                    group.remove_user_from_group(gitlab_user_id,gitlab_group_id)
                    module.exit_json(changed=True, result="Successfully removed user from the group '%s'." % gitlab_user)
        # user doesn't exist
        else:
            module.fail_json(msg="user not found '%s'." % gitlab_user)

    # group doesn't exist
    else:
        module.fail_json(msg="group not found  '%s'." % gitlab_group)
`

Does this make sense?

Reply:
NM. I have modified the code to have less if statements as it was suggested before. 
You may ignore this and let me know what you think about the new changes as you advised.

Reply:
Looks better, except for the point I originally pointed out in the above comment: if `state` is `absent`, you should not fail if the the user does not exist IMO.

Reply:
I added that point to the code in another file, which is different from the one I pushed!
This commit should have it.